pytorch_lightning/distributed/dist.py

```diff
@@ -33,3 +44,90 @@ def broadcast(self, obj: Any, group=_group.WORLD):
         broadcast_object_list(obj, 0, group=group or _group.WORLD)
 
         return obj[0]
+
+
+class SMLightningDistributed(LightningDistributed):
+
+    def broadcast(self, obj: Any, group=sm_dist.group.WORLD):
```

Comment:
`sm_dist` is conditionally imported. So this line fails when it's not present in cases where you train without that package.